src/algebra/module/pointwise_pi.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2021 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import algebra.pointwise
+import algebra.module.pi
+
+/-!
+# Pointwise actions on sets in Pi types
+
+This file contains lemmas about pointwise actions on sets in Pi types.
+
+## Tags
+
+set multiplication, set addition, pointwise addition, pointwise multiplication, pi
+
+-/
+
+open_locale pointwise
+open set
+
+@[to_additive]
+lemma smul_pi_subset {K R : Type*} [has_scalar K R] {ι : Type*} (r : K) (t : ι → set R) (s : set ι) :
```

Comment:
I think this is true for when `(t : Π (i : ι), set (α i))`, with dependent types, right?

Reply:
Yes, looks like everything in the file also works in that generality :)